pkg/dump/writer/tar_writer.go

```diff
@@ -87,24 +119,43 @@ func (t *TarWriter) Write(ctx context.Context, object []byte, filePath string) e
 // Flush function flushes all kubernetes object from the buffers to the tar file
 func (t *TarWriter) Flush(ctx context.Context) error {
 	log.I.Debug("Flushing writers")
-	span, _ := tracer.StartSpanFromContext(ctx, span.DumperWriterFlush, tracer.Measured())
-	span.SetTag(tag.DumperWriterTypeTag, TarTypeTag)
-	defer span.Finish()
-	for path, data := range t.buffers {
-		header := &tar.Header{
-			Name: path,
-			Mode: TarWriterChmod,
-			Size: int64(len(*data)),
-		}
+	t.mu.Lock()
+	defer t.mu.Unlock()
 
-		if err := t.tarWriter.WriteHeader(header); err != nil {
+	// Walking the memfs and copying all the files to the tar
+	err := afero.Walk(t.fs, "", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
 			return err
 		}
 
-		if _, err := t.tarWriter.Write(*data); err != nil {
-			return err
+		if info.IsDir() {
+			return nil
+		}
+		file, err := t.fs.Open(path)
```

Comment:
I think it may be worth putting everything after that line into a `writeTarArchiveFileToDisk(file)` or something along the line? What do you think?